Android/app/src/main/java/io/github/project_travel_mate/utilities/CurrencyActivity.java

```diff
@@ -0,0 +1,266 @@
+import butterknife.ButterKnife;
+import io.github.project_travel_mate.R;
+import io.github.project_travel_mate.destinations.description.WeatherActivity;
+import io.github.project_travel_mate.destinations.description.WeatherAdapter;
+import objects.CurrencyName;
+import objects.Weather;
+import okhttp3.Call;
+import okhttp3.Callback;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import utils.Compass;
+import utils.CurrencyConverterGlobal;
+
+import static utils.Constants.API_LINK_V2;
+import static utils.Constants.NUM_DAYS;
+import static utils.Constants.USER_TOKEN;
+import static utils.WeatherUtils.fetchDrawableFileResource;
+import static utils.WeatherUtils.getDayOfWeek;
+
+public class CurrencyActivity extends AppCompatActivity {
+    private static final String TAG = "CurrencyActivity";
+
+    @BindView(R.id.animation_view)
+    LottieAnimationView animationView;
+    @BindView(R.id.from_field)
+    RelativeLayout from_layout;
+    @BindView(R.id.to_field)
+    RelativeLayout to_layout;
+    @BindView(R.id.first_country_image)
+    ImageView from_image;
+    @BindView(R.id.second_country_flag)
+    ImageView to_image;
+    @BindView(R.id.first_country_edittext)
+    EditText from_edittext;
+    @BindView(R.id.text_result)
+    TextView result_textview;
+    @BindView(R.id.first_country_name)
+    TextView from_country_name;
+    @BindView(R.id.second_country_name)
+    TextView to_country_name;
+    @BindView(R.id.button_convert)
+    Button button_convert;
+
+    Boolean flag_check_first_item = false;
+    Boolean flag_check_second_item = false;
+    String from_amount = null;
+    String first_country_short;
+    String second_country_short;
+
+    private ProgressDialog mDialog;
+    public static ArrayList<CurrencyName> currences_names;
+
+    public static String sDefSystemLanguage;
+    private Context mContext;
+    private Handler mHandler;
+    private String mToken;
+
+    @Override
+    protected void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_utilities_currency_converter);
+        ButterKnife.bind(this);
+        setTitle("Currency Converter");
+        Objects.requireNonNull(getSupportActionBar()).setDisplayHomeAsUpEnabled(true);
+        getSupportActionBar().setHomeButtonEnabled(true);
+
+        SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(this);
+        mToken = sharedPreferences.getString(USER_TOKEN, null);
+        mDialog = new ProgressDialog(this);
+
+        sDefSystemLanguage = Locale.getDefault().getLanguage();
+
+        currences_names = new ArrayList<CurrencyName>();
+        mHandler = new Handler(Looper.getMainLooper());
+        mContext = this;
+
+        mDialog = new ProgressDialog(this);
+
+        first_country_short = "USD";
+        second_country_short = "INR";
+
+        from_layout.setOnClickListener(new View.OnClickListener() {
```

Comment:
You can use butterknife for this.

Reply:
These are just global variables. from_layout has it.